sapi/phpdbg/phpdbg_opcode.c

```diff
@@ -25,157 +25,18 @@
-		//zend_dump_class_fetch_type(op.num);
-	} else if (ZEND_VM_OP_CONSTRUCTOR == (flags & ZEND_VM_OP_MASK)) {
-		result = estrdup("CONSTRUCTOR");
-	}
-	return result;
-}
-
-char *phpdbg_decode_opline(zend_op_array *ops, zend_op *opline) /*{{{ */
-{
-	const char *opcode_name = phpdbg_decode_opcode(opline->opcode);
-	uint32_t flags = zend_get_opcode_flags(opline->opcode);
-	char *result, *decode[4] = {NULL, NULL, NULL, NULL};
-
-	/* OpcodeName */
-	if (opline->extended_value) {
-		spprintf(&decode[0], 0, "%s<%" PRIi32 ">", opcode_name, opline->extended_value);
-	}
-
-	/* OP1 */
-	decode[1] = phpdbg_decode_input_op(
-		ops, opline, opline->op1, opline->op1_type, ZEND_VM_OP1_FLAGS(flags));
-
-	/* OP2 */
-	decode[2] = phpdbg_decode_input_op(
-		ops, opline, opline->op2, opline->op2_type, ZEND_VM_OP2_FLAGS(flags));
-
-	/* RESULT */
-	switch (opline->opcode) {
-	case ZEND_CATCH:
-		if (opline->extended_value & ZEND_LAST_CATCH) {
-			if (decode[2]) {
-				efree(decode[2]);
-				decode[2] = NULL;
-			}
-		}
-		decode[3] = phpdbg_decode_op(ops, opline, &opline->result, opline->result_type);
-		break;
-	default:
-		decode[3] = phpdbg_decode_op(ops, opline, &opline->result, opline->result_type);
-		break;
-	}
-
-	spprintf(&result, 0,
-		"%-23s %-20s %-20s %-20s",
-		decode[0] ? decode[0] : opcode_name,
-		decode[1] ? decode[1] : "",
-		decode[2] ? decode[2] : "",
-		decode[3] ? decode[3] : "");
-
-	if (decode[0])
-		efree(decode[0]);
-	if (decode[1])
-		efree(decode[1]);
-	if (decode[2])
-		efree(decode[2]);
-	if (decode[3])
-		efree(decode[3]);
-
-	return result;
-} /* }}} */
-
 void phpdbg_print_opline_ex(zend_execute_data *execute_data, bool ignore_flags) /* {{{ */
 {
-	/* force out a line while stepping so the user knows what is happening */
-	if (ignore_flags ||
-		(!(PHPDBG_G(flags) & PHPDBG_IS_QUIET) ||
-		(PHPDBG_G(flags) & PHPDBG_IS_STEPPING) ||
-		(PHPDBG_G(oplog)))) {
-
-		zend_op *opline = (zend_op *) execute_data->opline;
-		char *decode = phpdbg_decode_opline(&execute_data->func->op_array, opline);
-
-		if (ignore_flags || (!(PHPDBG_G(flags) & PHPDBG_IS_QUIET) || (PHPDBG_G(flags) & PHPDBG_IS_STEPPING))) {
-			/* output line info */
-			phpdbg_notice("L%-5u %16p %s %s",
-			   opline->lineno,
-			   opline,
-			   decode,
-			   execute_data->func->op_array.filename ? ZSTR_VAL(execute_data->func->op_array.filename) : "unknown");
-		}
-
-		if (!ignore_flags && PHPDBG_G(oplog)) {
-			phpdbg_log_internal(fileno(PHPDBG_G(oplog)), "L%-5u %16p %s %s\n",
-				opline->lineno,
-				opline,
-				decode,
-				execute_data->func->op_array.filename ? ZSTR_VAL(execute_data->func->op_array.filename) : "unknown");
-		}
-
-		efree(decode);
+	if (ignore_flags || (!(PHPDBG_G(flags) & PHPDBG_IS_QUIET) && (PHPDBG_G(flags) & PHPDBG_IS_STEPPING))) {
+		fprintf(stderr, "[");
+		zend_dump_op_line(&EX(func)->op_array, NULL, EX(opline), ZEND_DUMP_LINE_NUMBERS | ZEND_DUMP_NO_LINE_END, NULL);
+		fprintf(stderr, "]\n");
```

Comment:
Do we want this to be wrapped in `[]`? This seems to be part of the phpdbg_notice formatting, but I'm not sure it makes sense here, now that this is no longer a "notice".

Reply:
I guess not.

6a3898195f29dd11b98f3a3f46d10c5ec732f881